POEM_035.md

```diff
@@ -0,0 +1,44 @@
+POEM ID: 035  
+Title: More generalized behavior in promoted inputs.  
+authors: [robfalck]  
+Competing POEMs: N/A  
+Related POEMs: N/A  
+Associated implementation PR:
+
+##  Status
+
+- [x] Active
+- [ ] Requesting decision
+- [ ] Accepted
+- [ ] Rejected
+- [ ] Integrated
+
+
+## Motivation
+
+The current input promotion system has exposed an inconsistency that can cause confusion among users and makes certain connections impossible.
+
+## Description
+
+Currently, if a variable is promoted using src_indices and it receives no connection, an automatic IndepVarComp output is created for it using those source indices to infer its shape.
+However, if an output is connected to it with some set of source indices, it currently generates an error.
+The `src_indices` from the connect call and the promotes call may conflict, and this is currently an error.
+This is a weakness in the current API.  A complex model may utilize src_indices deep within a nested system.
+A user may unwittingly try to connect to that with `src_indices` and encounter an error.
+
+![Current behavior for sizing a promoted input](POEM_035/current_behavior.png)
+
+Instead, this POEM proposes making src_indices apply to each level of promotion.
+During setup, OpenMDAO can then traverse the chain of connections and resolve which elements of the ultimate source end up where in the ultimate target.
+
+### Proposals
+
+1. `src_shape` is added as an argument to `promotes` and `add_subsystem` (when inputs are promoted), which specifies the apparent shape of the promoted input (if connecting to it) or the related AutoIVC output (if left unconnected).
+
+2. If `src_indices` are specified, then `src_shape` must be specified to remove any ambiguity, and `src_indices` are relative to that `src_shape`.
```

Comment:
This is backwards incompatible. Currently no user has specified src_shape. Did you mean it to be this way? 

Will we break old models?

Reply:
I think so but I also think there's no way to know the shape of the relevant input without tihs.  Rather than attempting to deduce this ourselves, I think it's best to require the user to be explicit about this.